### ISSUE TITLE
seq_bio losses too high

### ISSUE THREAD

**Issue by amunozo** [machamp-nlp/machamp#30]
I am training a multitask system with 3 tasks, 1 seq and 2 seq_bio. Although the scores seems to be similar, seq_bio losses are more than a thousand, while the seq is just 1. Is this normal?

**Comment by robvanderg** [comment]
In my (limited and biased) experience it is usually around 5-25 times larger; probably has to do with the difficulty of distinguishing boundaries, and that it does not use cross-entropy loss (like seq): https://github.com/machamp-nlp/machamp/blob/master/machamp/modules/allennlp/conditional_random_field.py#L322

1,000 seems quite excessive; are you training on publicly available datasets?

**Comment by amunozo** [comment]
I'm training for a shared-task on code-switching (https://codalab.lisn.upsaclay.fr/competitions/11030#learn_the_details). I tried before for seq and scores are similar, but loss is much lower. For training set, loss values range the values you said (5-25), but for the dev set they are over 2000.

**Comment by robvanderg** [comment]
Ah cool!, looks like an interesting task, was planning to participate but probably won't find the time. 

If the performance is in a sane range it sounds like there are just some cases where the loss explodes. You could try to inspect the values of: log_numerator - log_denominator in line https://github.com/machamp-nlp/machamp/blob/master/machamp/modules/allennlp/conditional_random_field.py#L342

What do the vocabularies for your NER task look like?, you can find them in the model directory in vocabularies/[task-name]/

**Comment by amunozo** [comment]
@@unkORpad@@
B-per
O
I-per
B-org
I-org
B-loc
I-loc
 
and
 
 @@unkORpad@@
O
B-ul
I-ul
B-cc
I-cc

**Comment by robvanderg** [comment]
Vocabularies look totally normal, and no huge label-space or anything that would explain it. 

Note that: log_numerator - log_denominator would be the losses per instance. You can match them with the `tags` variable in the same function, those are the gold labels. If you want to track them back to actual texts, you need the input_token_ids, which are only in `machamp.py` and can be converted to texts with AutoTokenizers convert_ids_to_tokens().

**Comment by amunozo** [comment]
Values of log_numerator - log_denominator are around minus thousands. What do you mean with matching them with the tags variables?

**Comment by robvanderg** [comment]
The `tags` variable that is used in the same function contains the gold labels; so if only some of them have extreme values, you could perhaps check which labels cause it (by finding the indices of the extreme ones in the `tags` variable. The main thing I am curious about is whether all of them have extreme values, or just some of them.

**Comment by amunozo** [comment]
Is it possible to link tags from one task to other during training? that is, task3 can only be X when task1 is Y

**Comment by robvanderg** [comment]
No, that is not supported

**Comment by amunozo** [comment]
After enough training, loss goes down in the trainin set. It just seems results are way sharper than for other tasks. However, scores are fairly low.

**Comment by robvanderg** [comment]
I am not sure what you mean by "sharper"?, ps. feel free to reach out on slack (https://join.slack.com/t/machamp-workspace/shared_invite/zt-1ln2ns7iv-A6nup3IrT3ZcOUuYAYU5Dw)  for a faster/more informal/les public discussion
